x-pack/plugins/lens/public/visualizations/partition/persistence.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cloneDeep } from 'lodash';
+import { LegendStats } from '@kbn/visualizations-plugin/common/constants';
+import { PieLayerState, PieVisualizationState } from '../../../common/types';
+
+type PersistedPieLayerState = Omit<PieLayerState, 'legendStats'> & {
+  showValuesInLegend?: boolean;
+};
+
+export type PersistedPieVisualizationState = Omit<PieVisualizationState, 'layers'> & {
+  layers: PersistedPieLayerState[];
+};
+
+export function convertToRuntime(state: PersistedPieVisualizationState) {
+  if (state.layers.some((l) => 'showValuesInLegend' in l)) {
+    return convertToLegendStats(state);
+  }
+  return state;
+}
```

Comment:
This does not allow for future runtime migrations to be added. I would suggest you clone the state at the top of this function and pass it to the `convertToLegendStats` to be mutated. Mutation is not ideal but the other options I see are passing the new state to each successive migration or worst of all, merging all the migrated states at the end.

```suggestion
export function convertToRuntime(state: PersistedPieVisualizationState) {
  const newState = cloneDeep(state);
  if (newState.layers.some((l) => 'showValuesInLegend' in l)) {
    convertToLegendStats(newState);
  }
  return newState;
}
```